### Ques 7. Using the data from Question 4, write code to analyze the data and answer the following questions

In [1]:
import pandas as pd
import warnings
warnings.filterwarnings("ignore")

In [2]:
df = pd.read_csv('Ques3.csv')

In [3]:
df.columns

Index(['name', 'id', 'nametype', 'recclass', 'mass', 'fall', 'year', 'reclat',
       'reclong', 'geolocation'],
      dtype='object')

### ● Get all the Earth meteorites that fell before the year 2000

In [4]:
print(df['year'][174])  # pd.dataframe has min limit
pd.Timestamp.min

1583-01-01


Timestamp('1677-09-21 00:12:43.145224193')

so we will change all str dates of the 'year' column to pd.datetime which will change all years below 1677 to pd.NaT (Not a Time)

In [5]:
dcopy=df.copy()
dcopy['year_only'] = df['year'] 
dcopy["year_only"] = pd.to_datetime(dcopy["year"], errors="coerce")

In [6]:
dcopy[dcopy['year_only'].apply(lambda x:x.year<2000)][['name','id','year']]

name     id        year
0       Aachen      1  1880-01-01
1       Aarhus      2  1951-01-01
2         Abee      6  1952-01-01
3     Acapulco     10  1976-01-01
4      Achiras    370  1902-01-01
..         ...    ...         ...
994   Timochin  24004  1807-01-01
995   Tirupati  24009  1934-01-01
997      Tjabe  24011  1869-01-01
998   Tjerebon  24012  1922-01-01
999  Tomakovka  24019  1905-01-01

[912 rows x 3 columns]

In [7]:
print(dcopy['year_only'][174])
print(dcopy['year_only'][174] is pd.NaT)

NaT
True


In [8]:
print(dcopy['year_only'][1])
print(dcopy['year_only'][1] is pd.NaT)

1951-01-01 00:00:00
False


### ● Get all the earth meteorites co-ordinates who fell before the year 1970

In [9]:
dcopy['coordinates']=dcopy['geolocation']

In [10]:
dcopy[dcopy['year_only'].apply(lambda x:x.year<1970)][['name','id','year','coordinates']]

name     id        year          coordinates
0       Aachen      1  1880-01-01      6.08333, 50.775
1       Aarhus      2  1951-01-01   10.23333, 56.18333
2         Abee      6  1952-01-01       -113, 54.21667
4      Achiras    370  1902-01-01    -64.95, -33.16667
5     Adhi Kot    379  1919-01-01           71.8, 32.1
..         ...    ...         ...                  ...
994   Timochin  24004  1807-01-01           35.2, 54.5
995   Tirupati  24009  1934-01-01   79.41667, 13.63333
997      Tjabe  24011  1869-01-01  111.53333, -7.08333
998   Tjerebon  24012  1922-01-01  106.58333, -6.66667
999  Tomakovka  24019  1905-01-01      34.76667, 47.85

[763 rows x 4 columns]

### ● Assuming that the mass of the earth meteorites was in kg, get all those whose mass was more than 10000kg

In [11]:
dcopy[dcopy['mass']>10000][['name','id','mass']]

name     id      mass
2          Abee      6  107000.0
7          Agen    392   30000.0
11          Aïr    424   24000.0
16      Akyumak    433   50000.0
27   Alfianello    466  228000.0
..          ...    ...       ...
991   Tieschitz  23989   28000.0
992      Tilden  23998   74800.0
994    Timochin  24004   65500.0
997       Tjabe  24011   20000.0
998    Tjerebon  24012   16500.0

[243 rows x 3 columns]